keras/layers/core/dense.py

```diff
@@ -331,6 +331,12 @@ def grad_fn(*args, upstream=None):
     def quantize(self, mode):
         import gc
 
+        # Prevent quantization of the subclasses
+        if type(self) is not Dense:
+            raise NotImplementedError(
```

Comment:
I just found that if downstream projects inherit these quantizable layers, quantization-related methods often fail because the weights and logic in `call` differ.

Therefore, `NotImplementedError` should be re-raised in `quantize` if these methods are not of these quantizable classes.

Reply:
For example, `keras_nlp.layers.ReversibleEmbedding` will call `quantize` and fail if we call `quantize` on the model including it.

Reply:
Got it, thanks.